op-e2e/e2eutils/geth/wait.go

```diff
@@ -104,3 +105,34 @@ func WaitForBlock(number *big.Int, client *ethclient.Client, timeout time.Durati
 		}
 	}
 }
+
+func WaitForBlockToBeFinalized(number *big.Int, client *ethclient.Client, timeout time.Duration) (*types.Block, error) {
+	return WaitForBlockTag(number, client, timeout, int64(rpc.FinalizedBlockNumber))
+}
+
+func WaitForBlockToBeSafe(number *big.Int, client *ethclient.Client, timeout time.Duration) (*types.Block, error) {
+	return WaitForBlockTag(number, client, timeout, int64(rpc.SafeBlockNumber))
+}
+
+// WaitForBlockTag polls for a block number to reach the specified tag & then returns that block at the number.
+func WaitForBlockTag(number *big.Int, client *ethclient.Client, timeout time.Duration, tag int64) (*types.Block, error) {
+	timeoutCh := time.After(timeout)
```

Comment:
I suspect this is copied from the existing `WaitForBlock` which uses a subscription and looks a bit suspiciously like it may have been written by some Java programmer that didn't get Golang contexts properly yet and looks a lot like me.....  I suspect we could clean up both.